tests/params/test_param.py

```diff
@@ -149,3 +149,67 @@ def test_dict_interface():
     assert list(param_store.keys()) == []
     assert list(key for key, value in param_store.items()) == []
     assert len(list(param_store.values())) == 0
+
+
+def test_scope():
```

Comment:
this doesn't seem to test the constraints?

Reply:
sure, new assertions added.